### PR TITLE
Add Custom Widget Indicator

### PR DESCRIPTION
**Pull request by OnderUlasKilic**
Added Custom widget indicator list tile

### CODE REVIEW

File: lib/101/custom_widget_learn.dart [hunk 1]
```diff
@@ -0,0 +1,74 @@
+// ignore_for_file: unused_import
```
Comment (BBarisKilic) [inline lib/101/custom_widget_learn.dart]: Remove this line since its unnecessary. (We should not keep anything that is not important in our project.)

**Comment by OnderUlasKilic** [comment]
Add Custom widget indicator list tile

File: lib/101/custom_widget_learn.dart [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+import 'package:flutter/material.dart';
+
+class CustomWidgetLearn extends StatelessWidget with _ColorsUtility {
+  CustomWidgetLearn({super.key});
```
Comment (BBarisKilic) [inline lib/101/custom_widget_learn.dart]: Add const here (Because all class variables are `final`)

File: lib/101/custom_widget_learn.dart [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  final String title = "Food";
```
Comment (BBarisKilic) [inline lib/101/custom_widget_learn.dart]: Replace `final String` with `static const`.

File: lib/101/custom_widget_learn.dart [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+mixin class _PaddingUtility {
+  final EdgeInsets normalPadding = const EdgeInsets.all(8.0);
+  final EdgeInsets normal2xPadding = const EdgeInsets.all(16.0);
+}
+
+class CustomFoodButton extends StatelessWidget
+    with _ColorsUtility, _PaddingUtility {
+  CustomFoodButton({super.key, required this.title, required this.onPressed});
```
Comment (BBarisKilic) [inline lib/101/custom_widget_learn.dart]: Add `const` keyword here to the constructor.

File: lib/101/custom_widget_learn.dart [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+mixin class _ColorsUtility {
+  final Color red = Colors.red;
+  final Color white = Colors.white;
+}
+
+mixin class _PaddingUtility {
+  final EdgeInsets normalPadding = const EdgeInsets.all(8.0);
+  final EdgeInsets normal2xPadding = const EdgeInsets.all(16.0);
+}
```
Comment (BBarisKilic) [inline lib/101/custom_widget_learn.dart]: (info) I assume you created these mixins for learning purposes, so its fine, but I would not suggest you to use this `with-mixin` approach in future projects.

File: lib/101/indicator_learn.dart [hunk 6]
```diff
@@ -0,0 +1,35 @@
+// ignore_for_file: unused_import
```
Comment (BBarisKilic) [inline lib/101/indicator_learn.dart]: Remove this line, it is not required.

**Review by BBarisKilic — CHANGES_REQUESTED** [review]
The new additions overall look fine 👍🏼

File: lib/101/custom_widget_learn.dart (hunk 5 above)
Comment (OnderUlasKilic) [reply lib/101/custom_widget_learn.dart]: Done.

File: lib/101/custom_widget_learn.dart (hunk 4 above)
Comment (OnderUlasKilic) [reply lib/101/custom_widget_learn.dart]: Done.

File: lib/101/custom_widget_learn.dart (hunk 3 above)
Comment (OnderUlasKilic) [reply lib/101/custom_widget_learn.dart]: Done.

File: lib/101/custom_widget_learn.dart (hunk 2 above)
Comment (OnderUlasKilic) [reply lib/101/custom_widget_learn.dart]: Done.

File: lib/101/custom_widget_learn.dart (hunk 1 above)
Comment (OnderUlasKilic) [reply lib/101/custom_widget_learn.dart]: Done.

**Review by BBarisKilic — APPROVED** [review]

**Pull request merged**